kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/PartitionUtils.java

```diff
@@ -509,4 +502,49 @@ protected static String serializePartitionValue(Literal literal) {
         }
         throw new UnsupportedOperationException("Unsupported partition column type: " + dataType);
     }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    // The following string escaping code is mainly copied from Spark                             //
+    // (org.apache.spark.sql.catalyst.catalog.ExternalCatalogUtils) which is copied from          //
+    // Hive (o.a.h.h.common.FileUtils).                                                           //
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    private static final BitSet CHARS_TO_ESCAPE = new BitSet(128);
+
+    static {
+        // ASCII 01-1F are HTTP control characters that need to be escaped.
+        char[] controlChars = new char[] {
+            '\u0001', '\u0002', '\u0003', '\u0004', '\u0005', '\u0006', '\u0007', '\b',
+            '\t', '\n', '\u000B', '\f', '\r', '\u000E', '\u000F', '\u0010', '\u0011',
+            '\u0012', '\u0013', '\u0014', '\u0015', '\u0016', '\u0017', '\u0018', '\u0019',
+            '\u001A', '\u001B', '\u001C', '\u001D', '\u001E', '\u001F', '"', '#', '%', '\'',
+            '*', '/', ':', '=', '?', '\\', '\u007F', '{', '[', ']', '^'
+        };
+
+        for (char c : controlChars) {
+            CHARS_TO_ESCAPE.set(c);
+        }
+    }
+
+    /**
+     * Escapes the given string to be used as a partition value in the path. Basically this escapes
+     * - characters that can't be in a file path. E.g. `a\nb` will be escaped to `a%0Ab`. -
```

Comment:
what's up with the `-` usage in this comment? is it a dot job? the "-" literal character?

Reply:
auto-format. changed it to use the proper lists`<ul>`